grype/pkg/upstream_package.go

```diff
@@ -0,0 +1,58 @@
+package pkg
+
+import (
+	"strings"
+
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/scylladb/go-set/strset"
+)
+
+type UpstreamPackage struct {
+	Name    string // the package name
+	Version string // the version of the package
+}
+
+func UpstreamPackages(p Package) <-chan Package {
```

Comment:
Is there a reason for this being `<-chan Package` instead of, say, `[]Package`?

Reply:
There is no strong reason for this --no. The current approach is more like a python generator, but I can change it to return a static slice too 👍